test/fixtures/es-module-loaders/hooks-input.mjs

```diff
@@ -17,7 +17,11 @@ export async function resolve(specifier, context, next) {
   if (resolveCalls === 1) {
     url = new URL(specifier).href;
     assert.match(specifier, /json-modules\.mjs$/);
-    assert.strictEqual(context.parentURL, undefined);
+
+    if (context.parentURL && !context.parentURL.includes('eval')) {
+      assert.strictEqual(context.parentURL, undefined);
+    }
```

Comment:
If `context.parentURL` is truthy, there's no way it can be undefined, the assert is not reachable